### PR TITLE
I18n Addon

### PR DESCRIPTION
**Pull request by Luke-Larsen**
How the system works.
1) During the loading of the webpage a small JS script at line 50 on index.html checks what language the users browser is, it then checks if there is an active language file in the array activeLangs. This check is in place so that some language files that haven't been updated to new code could be disabled if there are major changes. Then it writes a script tag to include that file, if the language isn't active yet, then it defaults back to {en-US}. 

2) On line 721 of the index.html It calls the translation script from /scripts/i18n.js. This script reads through the html file and looks for data-i18n information in the tags and replaces the text with what is in the translation.js object. If the object doesn't have something it leaves it as the default. The translation.js allows for simple HTML tags like b and i. It is sanitized by element.innerHTML in the browser.

### CODE REVIEW

File: locale/en-US/translation.js
Comment (JSKitty) [inline locale/en-US/translation.js:120]: We should keep any HTML out of here to simplify it for translators (would simply concatenate the HTML elements around the translation string).

**Review by JSKitty — CHANGES_REQUESTED** [review]
I've done some *light* cleanup and functionality improvements, and there's more to do; but overall I'm **really** liking the PR :purple_heart: **super** lightweight system and I think that's for the best :rocket: we don't need anything too heavy-weight, and as long as it's maintainable somewhat (perhaps we could write some local tooling for this *eventually?*), it seems good to me. :tada: 

:bulb: Since we have an array of supported languages, I suggest we add the necessary UI in the Settings to 'swap' languages, since one's browser locale may not always be accurate (for privacy) or in other cases users simply want to see MPW in their chosen lang.

A dropdown (much like our Explorer one) would be perfect and easy to implement, dump it to LocalStorage for persistence, and we're good to roll.

**Review by JSKitty — COMMENTED** [review]
Oh, and when this is merged, I'm absolutely adding a stupid language like "Pirate" or "Cat" language.

File: index.html
Comment (Duddino) [inline index.html:582]: Rewards isn't being translated

File: locale/en/translation.js
Comment (Duddino) [inline locale/en/translation.js:123]: Looks like "sendSimpleTransactionTitle" is missing from this file

File: index.html
Comment (Duddino) [inline index.html:328]: This isn't being translated

**Review by Duddino — CHANGES_REQUESTED** [review]
Looks really good to me, although some things aren't being translated.
Also check things like index.html:826 where divs are added through javascript

File: index.html
Comment (Duddino) [reply index.html:335]: I mean line 328

**Review by Duddino — CHANGES_REQUESTED** [review]
Really liking the work so far!
The only suggestion, other than minor style nitpicks, is that the alert translation system could be extended to all dynamic html code. I would prefer a function that accepted a string, so code like
```js
return `Please confirm this is the address you see on your ${strHardwareName}.
           <div class="seed-phrase">${address}</div>`;
```
could become
```js
return translateText("address-confirmation", strHardwareName, address);
```
Not sure how all of this could be dynamically translated though

**Comment by Duddino** [comment]
Sorry, by "dynamic translated" I meant updated when you switch language, we should totally stick with this style of translation

**Comment by JSKitty** [comment]
The nature of this PR / i18n makes it almost impossible to truly "finalise" (there are **constant** UI updates, rewrites, word changing and more) as such I'm happy to merge this PR exclusively as long as "existing" translations and features are working.

We can handle "catching up" with UI changes and additions in consecutive PRs, and translators can also do the same, missed or old translations will just be replaced with the English version, worst case scenario.

There are two conflicts currently (as GitHub shows above), once these are resolved, I'll give this PR my **tACK**.

**Review by JSKitty — DISMISSED** [review]
**tACK**, I've done some cleanup and also improved Alerts functionality a bit, this seems easy enough to maintain, and missing translations + template can be slowly build upon in future PR iterations.

Static texts, alerts, with/without LocalStorage, all working fine for me.

**Review by panleone — CHANGES_REQUESTED** [review]
Good work but in my opinion there are still many things to change:
1) With the current system there is NO WAY to translate phrases like this
https://github.com/PIVX-Labs/MyPIVXWallet/blob/711eae6a0b6dcdc3c90d954b985f57da381ae431/index.html#L1281
2) It is not well merged: for example the masternode tab disappeared

**Review by JSKitty — COMMENTED** [review]
Two things:
- I have applied a patch to `createAlert` for maintainer Quality-of-Life, although it could be considered hacky, since this PR replaces the parameter position of `timeout` with `alertVariables`, it forces us to either: add an empty `[]` to every alert (annoying), or to do what I've done in my last commit, which I feel is better.

- The majority of the master-branch merge went A-OK :pinched_fingers: it seems, but I notice in the Settings, our "Node" setting (which should show Duddino's RPC node) has disappeared and is unselectable.

**Review by JSKitty — COMMENTED** [review]
Alerts + Popups + Static translations all seem to be working perfectly as expected. :rocket: :tada: 

There may be some small merge issues (or undetected code issues) though, as I've noticed the "PIVX Node" setting is bugged, blank without any selections.
image

**Review by 11Tuvork28 — CHANGES_REQUESTED** [review]
Could you add JDOC comments to newly introduced functions?
Would be nice

**Review by 11Tuvork28 — DISMISSED** [review]
LGTM

**Review by JSKitty — APPROVED** [review]
**tACK**, I can't find any issues with existing translations, nor any merge bugs with Master branch, I have tried quite a few things and I'm not able to find **anything** now to complain about.

LGTM before this gets buried in further conflicts.

File: scripts/i18n.js
Comment (Duddino) [inline scripts/i18n.js:7]: This was never declared

File: scripts/i18n.js
Comment (Duddino) [inline scripts/i18n.js:1]: Add "use strict";

**Review by Duddino — DISMISSED** [review]
Other than the two minor nitpicks, I think the PR is ready to be merged. Some things haven't been translated yet, but I think it's better to split them up in different PRs to prevent massive conflicts

**Review by panleone — APPROVED** [review]
tACK everything is working as it should

**Review by Duddino — APPROVED** [review]
tACK maybe we can finally merge this

**Review by JSKitty — APPROVED** [review]
**tACK**, finally, this damn thing is getting merged, and most importantly we support **uwu** language.

**Comment by JSKitty** [comment]
:coin: **1,750 PIV** bounty fulfilled and sent

**Pull request merged**
